### PR TITLE
Reduce BaseElement interface

### PR DESCRIPTION
**Pull request by rocky**
* move round_to_float() into NumericOperations
* replace x.is_atom() with isinstance(x, Atom)
* replace x.is_symbol() with isinstance(x, Symbol)

As before, these mechanical changes touches lots of files.  (Not something I am fond of doing, gratuitously, were it not for the fact it was holding me up from more important things - see below).

Next up - splitting off BaseElement from mathics.core.symbols into a new module matthics.core.element

I had originally wanted to do this first, but because of the modularity violations in BaseElement I couldn't and had to do all of these others _first_. (I think the one that was crucial was the work to NumericOperations. But I couldn't have know that previously)

### CODE REVIEW

File: mathics/core/symbols.py [hunk 1]
```diff
@@ -123,7 +146,7 @@ def __pow__(self, other) -> "BaseElement":
 # to be intercombined here.
 
 
-class KeyComparable(object):
+class KeyComparable:
```
Comment (rocky) [inline mathics/core/symbols.py:149]: @TiagoCavalcante a historical note: the reason `object `was there originally is that way back when this code was written (but the change to Python went in around 2.4 or so) there was skill a notion of the the kind of Python lame "class" that Python originally had.  

It always amazed me how long it too Python to fix things that started out doing brokenly but not were broken in other contemporaneous languages.

Back then, in order to get the current style class you had to use this `X(object)` form. 
https://stackoverflow.com/questions/54867/what-is-the-difference-between-old-style-and-new-style-classes-in-python

At some point, probably Python 3 the non-broken from of class became the default. But I think in Python 2.7 that wasn't the default. 

These are just stories you can tell your grandchildren.

Comment (rocky) [reply mathics/core/symbols.py:149]: Oh - more stories you can tell your grandchildren about the good old days. 

Back then there was no notion of a "folder" or "directory" so all programs had to put in one "file" which was called "a deck of cards".  So Originally Mathics was one file. 

And this was before graph theory was invented and they were still trying to figure out whether all the nodes in a tree should be called "leaves"

**Review by TiagoCavalcante — APPROVED** [review]
LGTM

File: mathics/core/symbols.py (hunk 1 above)
Comment (TiagoCavalcante) [reply mathics/core/symbols.py:149]: > @TiagoCavalcante a historical note: the reason `object `was there originally is that way back when this code was written (but the change to Python went in around 2.4 or so) there was still a notion of the the kind of Python lame "class" that Python originally had.
> 
> It always amazed me how long it took Python to fix things that started out doing brokenly but not were broken in other contemporaneous languages.
> 
> Back then, in order to get the current style class you had to use this `X(object)` form. https://stackoverflow.com/questions/54867/what-is-the-difference-between-old-style-and-new-style-classes-in-python
> 
> At some point, probably Python 3 the non-broken from of class became the default. But I think in Python 2.7 that wasn't the default.
> 
> These are just stories you can tell your grandchildren.

Oh, I didn't know it, I thought that in Python 2 every class needed to extend something.

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**